test/functional/isVisible.js

```diff
@@ -12,45 +12,46 @@ describe('isVisible', () => {
     const options = new firefox.Options();
     options.headless();
 
-    const driver = new webdriver.Builder()
+    const driver = new Builder()
         .forBrowser('firefox')
         .setFirefoxOptions(options)
         .build();
 
-    it('Should return false when an element is hidden', async () => {
-        const hiddenElementIds = [
-            'not-visible-1',
-            'not-visible-2',
-            'not-visible-3',
-            'not-visible-4',
-            'not-visible-5',
-            'not-visible-6',
-            'not-visible-7',
-            'not-visible-8',
-            'not-visible-9',
-            'not-visible-10',
-            'not-visible-11',
-            'not-visible-12',
-        ];
+    async function checkVisibility(id, expected) {
+        await driver.wait(until.elementLocated(By.id(id)), WAIT_MS);
+        const isElementVisible = await driver.executeScript(`
+            ${ancestors}
+            ${isDomElement}
+            ${toDomElement}
+            return ${isVisible}(document.getElementById('${id}'));
+        `);
+        assert.equal(
+            isElementVisible,
+            expected,
+            `isVisible should return false for element with id '${id}'.`
+        );
+    }
+
+    it('should return false when an element is hidden', async () => {
+        const hiddenElementIds = await driver.executeScript(`
+            return [].slice.call(document.querySelectorAll('[id^="not-visible-"]')).map((element) => element.id);
```

Comment:
Is the `slice` just to copy the results of querySelectorAll into an Array? If so, Array.from(querySelectorAll(...)) might show your intention more clearly.

Shorter still, you could do `Array.prototype.map.call(querySelectorAll(...), ...)`.